pkg/cmd/cli/cmd/rsync/rsync.go

```diff
@@ -229,7 +236,76 @@ func (o *RsyncOptions) Validate() error {
 
 // RunRsync copies files from source to destination
 func (o *RsyncOptions) RunRsync() error {
-	return o.Strategy.Copy(o.Source, o.Destination, o.Out, o.ErrOut)
+	err := o.Strategy.Copy(o.Source, o.Destination, o.Out, o.ErrOut)
+	if err != nil || !o.Watch {
+		return err
+	}
+
+	dirty := false
+	lastChange := time.Now()
+	var watchError error
+	changeLock := sync.Mutex{}
```

Comment:
Please document what the mutex is to be used for. It protect access to `dirty`, `lastChange` and `watchError`, right?
Whoever will be reading/changing this code in the future will need to keep that in mind.

This is more of a stylistic decision, but all those four things could be declared in a single `var` block with documentation comments (similar to what we do in `**/api/**/types.go` files), and rely on the zero values for the respective types:

```go
// blah blah blah... changeLock synchronizes changes of the state (dirty, lastChange, watchError)
// and should be acquired whenever reading or writing one of those variables...
var (
  dirty      bool
  lastChange time.Time
  watchError error
  changeLock sync.Mutex
)
```